src/Tasks/Copy.cs

```diff
@@ -288,6 +319,7 @@ private void LogAlwaysRetryDiagnosticFromResources(string messageResourceName, p
             }
 
             if (!Traits.Instance.EscapeHatches.CopyWithoutDelete &&
+                (UseHardlinksIfPossible || UseSymboliclinksIfPossible) &&
```

Comment:
This seems to break tests. Why is it beneficial?

Reply:
My intent was to try and minimize deletes when the copy would overwrite it anyway. Going to undo this change.